src/h5web/vis-packs/core/shared/VisCanvas.tsx

```diff
@@ -14,8 +15,7 @@ interface Props {
 function VisCanvas(props: Props) {
   const { axisOffsets, children, aspectRatio } = props;
 
-  const [visAreaRef, visAreaSize] = useMeasure();
-
+  const [visAreaSize = EMPTY_RECT, visAreaRef] = useMeasure<HTMLDivElement>();
```

Comment:
Instead of using `visAreaSize`, you could add an early return here:

```js
if(!visAreaSize) {
      return (<div ref={visAreaRef} className={styles.visArea} />);
}
```

We would avoid computing `availableSize` and `visSize` for nothing below.

Reply:
seems legit

Reply:
I removed `EMPTY_RECT` in the end (cf. `ColorBar.tsx`)